dbdimp.c

```diff
@@ -3657,8 +3661,10 @@ int dbd_st_execute(SV* sth, imp_sth_t* imp_sth)
       /** Store the result in the current statement handle */
       DBIc_NUM_FIELDS(imp_sth)= mysql_num_fields(imp_sth->result);
       DBIc_ACTIVE_on(imp_sth);
+#if MYSQL_VERSION_ID >= SERVER_PREPARE_VERSION
       if (!imp_sth->use_server_side_prepare)
         imp_sth->done_desc= 0;
```

Comment:
I think you want to set done_desc to 0 even if server side prepare statements are not supported by mysql client...